### PR TITLE
sbt-ci-build-doc.bsh

### PR DESCRIPTION
**Pull request by wrouesnel**
Adds the new general purpose sbt build-doc script to support integration with GHE and drone.

This requires the accompanying modification to `uniform` - https://github.com/CommBank/uniform/pull/78

This needs some careful validation, posted here for initial review.

### CODE REVIEW

**Comment by wrouesnel** [comment]
I'm totally punting on solving the git credential distribution problem in this script. The exact argument being made is "the remote we extract from `origin` should have been setup in advance to provide access".

There's no general purpose way to do it that'll work across all our systems - SSH keys come closest, as Drone automatically sets one up for the repository it pulls from, and Travis supports an OOB injection mechanism (both of which would accomplish the goal of definitely not leaking credentials).

File: sbt-ci-build-doc.bsh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    echo "src/site/_config.yml does not exist. Aborting."
+    exit 1
+fi
+
+echo "" >> src/site/_config.yml
+echo "releaseVersion: $version" >> src/site/_config.yml
+
+# This command dumps the documentation into target/site
+#echo "Creating site..."
+#sbt -Dsbt.global.base=$CI_BUILD_DIR make-site
```
Comment (stephanh) [inline sbt-ci-build-doc.bsh]: Commented out code?

File: sbt-ci-build-doc.bsh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#   Usage:
+#   ./ci/sbt-ci-build-doc <documentation root URL> \
+#       <documentation source code URL base>
+#
+#   Example:
+#   ./ci/sbt-ci-build-doc https://commbank.github.io/ \
+#       https://github.com/CommBank/maestro/ \
```
Comment (stephanh) [inline sbt-ci-build-doc.bsh]: Are you sure we need the `maestro` part at the end?

File: sbt-ci-build-doc.bsh [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    fi
+}
+
+if [ $CI_BRANCH = "master" ]; then
+    do_build_doc
+    if [ -z $DOCS_NOPUBLISH ]; then
+        do_publish  
+    fi
+else
+    if [ ! -z $FORCE_DOCS ]; then
```
Comment (stephanh) [inline sbt-ci-build-doc.bsh]: Can you document some of these variables that modify behaviour.

Also to be consistent I think you should change `FORCE_DOCS` to `DOCS_FORCE` then it fits in nicely with your other `DOCS_` prefixed envs.

**Comment by stephanh** [comment]
:+1:

**Comment by wrouesnel** [comment]
As agreed the difficulty of testing this properly at the moment means I'm going to abstract the code for dockermake, then revisit and hopefully have some new tests for this.

**Comment by wrouesnel** [comment]
Closing this - it'll be superceded by a new PR.

**Pull request closed without merging**